datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -300,6 +306,130 @@ fn cast_static_filter_to_set(list: &[Arc<dyn PhysicalExpr>]) -> HashSet<ScalarVa
     }))
 }
 
+fn make_list_contains_decimal(
+    array: &DecimalArray,
+    list: Vec<ColumnarValue>,
+    negated: bool,
+) -> BooleanArray {
+    let contains_null = list
+        .iter()
+        .any(|v| matches!(v, ColumnarValue::Scalar(s) if s.is_null()));
+    let values = list
+        .iter()
+        .flat_map(|v| match v {
+            ColumnarValue::Scalar(s) => match s {
+                Decimal128(v128op, _, _) => v128op.map(|v128| v128),
+                _ => {
+                    unreachable!(
+                        "InList can't reach other data type for decimal data type."
+                    )
+                }
+            },
+            ColumnarValue::Array(_) => {
+                unimplemented!("InList does not yet support nested columns.")
+            }
+        })
+        .collect::<Vec<_>>();
+
+    if !negated {
+        // In
+        array
+            .iter()
+            .map(|v| match v {
+                None => None,
+                Some(v128) => {
+                    if values.contains(&v128) {
+                        Some(true)
+                    } else {
+                        Some(false)
+                    }
+                }
+            })
+            .collect::<BooleanArray>()
+    } else {
+        // Not in
+        if contains_null {
+            // if the expr is NOT IN and the list contains NULL value
+            // All the result must be NONE
+            vec![None; array.len()]
+                .into_iter()
+                .collect::<BooleanArray>()
+        } else {
+            array
+                .iter()
+                .map(|v| match v {
+                    None => None,
+                    Some(v128) => {
+                        if !values.contains(&v128) {
+                            Some(true)
+                        } else {
+                            Some(false)
+                        }
+                    }
+                })
+                .collect::<BooleanArray>()
+        }
+    }
+}
+
+fn make_set_contains_decimal(
```

Comment:
The same comments apply here as do `make_list_contains_decimal` as they are largely copy/paste of each other.